lib/ex_webrtc/peer_connection.ex

```diff
@@ -470,91 +457,146 @@ defmodule ExWebRTC.PeerConnection do
-    # TODO: at the moment, we assign mids in create_offer
-    transceivers
+      state = set_description(:local, type, sdp, state)
+      {:ok, %{state | signaling_state: next_sig_state}}
+    end
   end
 
-  defp update_local_transceivers(:answer, transceivers, _sdp) do
-    transceivers
-  end
+  defp apply_remote_description(%SessionDescription{type: type}, _state)
+       when type in [:rollback, :pranswer],
+       do: {:error, :not_implemented}
 
-  defp apply_remote_description(type, sdp, state) do
-    # TODO apply steps listed in RFC 8829 5.10
-    with :ok <- SDPUtils.ensure_mid(sdp),
+  defp apply_remote_description(%SessionDescription{type: type, sdp: raw_sdp}, state) do
+    with {:ok, next_sig_state} <- next_signaling_state(state.signaling_state, :remote, type),
+         {:ok, sdp} <- parse_sdp(raw_sdp),
+         :ok <- SDPUtils.ensure_mid(sdp),
          :ok <- SDPUtils.ensure_bundle(sdp),
+         :ok <- SDPUtils.ensure_rtcp_mux(sdp),
          {:ok, {ice_ufrag, ice_pwd}} <- SDPUtils.get_ice_credentials(sdp),
          {:ok, {:fingerprint, {:sha256, peer_fingerprint}}} <- SDPUtils.get_cert_fingerprint(sdp),
-         {:ok, new_transceivers} <-
-           update_remote_transceivers(state.transceivers, sdp, state.config) do
-      :ok =
-        state.ice_transport.set_remote_credentials(state.ice_pid, ice_ufrag, ice_pwd)
-
-      :ok = state.ice_transport.gather_candidates(state.ice_pid)
-
-      # TODO: this needs a look
-
-      new_remote_tracks =
-        new_transceivers
-        # only take new transceivers that can receive tracks
-        |> Enum.filter(fn tr ->
-          RTPTransceiver.find_by_mid(state.transceivers, tr.mid) == nil and
-            tr.direction in [:recvonly, :sendrecv]
-        end)
-        |> Enum.map(fn tr -> tr.receiver.track end)
-
-      for track <- new_remote_tracks do
-        notify(state.owner, {:track, track})
-      end
+         {:ok, dtls_role} <- SDPUtils.get_dtls_role(sdp),
+         {:ok, transceivers} <- update_transceivers(state, sdp) do
+      :ok = state.ice_transport.set_remote_credentials(state.ice_pid, ice_ufrag, ice_pwd)
 
-      state = set_description(:remote, type, sdp, state)
+      for candidate <- SDPUtils.get_ice_candidates(sdp) do
+        state.ice_transport.add_remote_candidate(state.ice_pid, candidate)
+      end
 
-      if type == :answer do
-        {:setup, setup} = ExSDP.Media.get_attribute(hd(sdp.media), :setup)
+      # infer our role from the remote role
+      dtls_role = if dtls_role == :active, do: :passive, else: :actice
+      DTLSTransport.start_dtls(state.dtls_pid, dtls_role, peer_fingerprint)
 
-        setup =
-          case setup do
-            :active -> :passive
-            :passive -> :active
-          end
+      state = %{state | demuxer: Demuxer.update(state.demuxer, sdp)}
 
-        DTLSTransport.start_dtls(state.dtls_transport, setup, peer_fingerprint)
-      end
+      # TODO: for now, we emit track event for every new transceiver
```

Comment:
For now? Is this subject to change?

Reply:
Renegotation can result in a new track on existing transceiver and we don't handle that. That's what I meant.

Reply:
Could we write this in the comment?